src/utils/components/SidebarEditor/SidebarEditor.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { dump } from 'js-yaml';
+
+import { K8sResourceCommon, YAMLEditor } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  Flex,
+  FlexItem,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+
+import Loading from '../Loading/Loading';
+
+import { SidebarEditorContext } from './SidebarEditorContext';
+import { safeLoad } from './utils';
+
+import './sidebar-editor.scss';
+
+type SidebarEditorProps<Resource> = {
+  resource: Resource;
+  onResourceUpdate?: (newResource: Resource) => Promise<Resource | void>;
+  children: (resource: Resource) => React.ReactNode;
+};
+
+const SidebarEditor = <Resource extends K8sResourceCommon>({
+  children,
+  resource,
+  onResourceUpdate,
+}: SidebarEditorProps<Resource>): JSX.Element => {
+  const resourceYAML = React.useMemo(() => dump(resource, { skipInvalid: true }), [resource]);
+  const { showEditor } = useContext(SidebarEditorContext);
+  const [editableYAML, setEditableYAML] = useState('');
+  const editedResource = safeLoad<Resource>(editableYAML);
+
+  const reloadResource = useCallback(() => {
+    setEditableYAML(resourceYAML);
+  }, [resourceYAML]);
+
+  useEffect(() => {
+    reloadResource();
+  }, [reloadResource]);
+
+  const changeResource = (newValue: string) => {
+    setEditableYAML(newValue);
+
+    return {};
+  };
+
+  const edited = editableYAML !== resourceYAML;
+
+  return (
+    <Sidebar isPanelRight hasGutter hasNoBackground className="sidebar-editor">
+      <SidebarContent>{children && children(editedResource ?? resource)}</SidebarContent>
+      {showEditor && (
+        <SidebarPanel
+          width={{ default: 'width_33', lg: 'width_50', xl: 'width_50' }}
+          className="sidebar-editor"
+        >
+          <Stack hasGutter>
+            <StackItem isFilled>
+              <React.Suspense fallback={<Loading />}>
+                <YAMLEditor value={editableYAML} minHeight="300px" onChange={changeResource} />
```

Comment:
can we set min height via classname?

Reply:
no className props. minHeight is used for inline style in the source code combined with react-measure. But I can do it with the parent